src/networkpolicy/deduplicator.go

```diff
@@ -764,8 +765,14 @@ func existPolicyName(policyNamesMap map[string]bool, name string) bool {
 
 func GeneratePolicyName(policyNamesMap map[string]bool, policy types.KnoxNetworkPolicy, clusterName string) types.KnoxNetworkPolicy {
 	polType := policy.Metadata["type"]
+	labels := []string{}
 
-	hashInt := common.HashInt(polType+policy.Metadata["labels"]+policy.Metadata["namespace"]+policy.Metadata["clustername"]+policy.Metadata["containername"])
+	for k, v := range policy.Spec.Selector.MatchLabels {
+		labels = append(labels, k+"="+v)
+	}
+	sort.Strings(labels) // sort labels
+
+	hashInt := common.HashInt(polType + strings.Join(labels, ",") + policy.Metadata["namespace"] + clusterName + policy.Metadata["container_name"])
```

Comment:
Why do we have to include `container_name` to generate the hash? In case of network-policy, we can only apply policy for a deployment/pod [labels + namespace]. If we have a multi-contianer pod, including `container_name`, will generate multiple policies for the same pod/deployment.